### ISSUE TITLE
Turn hilt-extensions into regular java library

### ISSUE THREAD

**Issue by matejdro** [sczerwinski/android-hilt#384]
Currently, `hilt-extensions` is an android library, which prevents it from being used in non-android modules.

From what I can see, this library only contains a bunch of annotations, it has no Android references. Could we turn it into regular java library?

**Comment by sczerwinski** [comment]
These extensions could be useful for Dagger projects as well. Unfortunately, I'm a bit busy with other work right now.

Would you like to create a pull request with these changes, @matejdro?

**Comment by matejdro** [comment]
Yes, I can try my hand at the pull request. But I can't promise on any dates, it might take a while.
